### PR TITLE
Updated Functions | Easier Logout | Easier Login

### PR DESCRIPTION
**Pull request by RedSparr0w**
made demo.php file a bit tidier and easier to read

Also added check for `!isset($_GET['login'])` as to avoid having to refresh on login

### CODE REVIEW

File: example.php [hunk 1]
```diff
@@ -12,12 +12,13 @@
...
 <body>
 <?php
-if(!isset($_SESSION['steamid'])) {
+if(!isset($_SESSION['steamid']) && !isset($_GET['login'])) {
 
     echo "welcome guest! please login \n \n";
     steamlogin(); //login button
     
 }  else {
+    steamlogin(); //login
```
Comment (BlackCetha) [inline example.php]: This doesnt make sense at all.
Youre displaying a login button even if the user is logged in.

File: demo.php [hunk 2]
```diff
@@ -27,29 +27,111 @@
 		<span class="small pull-left" style="padding-right: 10px;">for SteamAuth 2.1.2</span>
 		<hr>
 		<?php
-if(!isset($_SESSION['steamid'])) {
+if(!isset($_SESSION['steamid']) && !isset($_GET['login'])) {
```
Comment (BlackCetha) [inline demo.php]: How exactly would this check remove the need to refresh?

**Comment by RedSparr0w** [comment]
if you look at the function of `steamlogin()` it does not show the login button if logging the userin https://github.com/SmItH197/SteamAuthentication/blob/master/steamauth/steamauth.php#L21-L27

**Comment by RedSparr0w** [comment]
Updated

Makes more sense now. 
If `$_GET['login']` is set and user not logged in already (incase refresh page with params still there) then this will log user in and not display a blank page with "Welcome Guest! Please log in!"

**Comment by RedSparr0w** [comment]
Sorry, Had not set a "loginpage" hence the refresh problem. XD

Have updated "settings.php" page so when you login, it will redirect you back to the page you clicked the login button

Also Tidied up "demo.php" page for easier readability

**Comment by BlackCetha** [comment]
Did you test it?
AFAIK `PHP_SELF` will be either `/steamauth/settings.php` or `/` because of the redirection from steam.
Also, what exactly is your "refresh problem"?
While the changes to the demo make sense the one in the settings doesnt.

Please split them up at least. 2 Features/Changes = 2 PRs.

**Comment by RedSparr0w** [comment]
- steam redirects back to the origin page so `PHP_SELF` is the origin page ie `/demo.php` (doesnt refer to `/steamauth/settings.php` page) - have tested with demo page and example page
- the "refresh problem" i was getting was because no `$steamauth['loginpage']` was specified so was not redirecting/reloading after signin
- no problem will split it into 2 different PRs now

**Comment by RedSparr0w** [comment]
Have updated now. 

Changed they way i did it so "settings.php" file remains unchanged but updated the `steamlogin()` function so if `$steamauth['loginpage']` is not set to a specific page in the settings then it will reload the page you are currently on.

**Comment by BlackCetha** [comment]
Seems fine now.
I actually made the same change a few month ago, but we made a rollback because there were too many bugs in a bunch of commits.

**Comment by RedSparr0w** [comment]
just had a look

```
if($steamauth['loginpage'] !== "") {
	$returnTo = $steamauth['loginpage'];
} else {
	//Determine the return to page. We substract "login&"" to remove the login var from the URL.
	//"file.php?login&foo=bar" would become "file.php?foo=bar"
	$returnTo = str_replace('login&', '', $_GET['openid_return_to']);
	//If it didn't change anything, it means that there's no additionals vars, so remove the login var so that we don't get redirected to Steam over and over.
	if($returnTo === $_GET['openid_return_to']) $returnTo = str_replace('?login', '', $_GET['openid_return_to']);
}
```

not sure why that wouldn't have worked. 
only thing i can see is `$returnTo = str_replace('login&', '', $_GET['openid_return_to']);` wouldnt do anything due to if there was a "&" in the return url it would be interpreted as a new `$_GET` param in the current url and therefore disregarded/not included in the `$_GET['openid_return_to']` param? 

this may also work instead?

```
if (empty($steamauth['loginpage']))
	$steamauth['loginpage'] = str_replace('?login', '', $_GET['openid_return_to']);
}
header('Location: '.$steamauth['loginpage']);
```

**Comment by BlackCetha** [comment]
Steam strips any GET-parameters from the return-url automatically, so it didnt work.
After that we tried to get it to work in some way, but it only made things worse.

**Comment by RedSparr0w** [comment]
i feel like my changes should work, have tested it out on a localserver and c9 server and my own site, all seemed to work fine.

**Comment by RedSparr0w** [comment]
https://github.com/RedSparr0w/SteamAuthentication/commit/77f02582a82a8c911ab9c6d08adeff3a3a54d2c8 fixes #107

**Comment by RedSparr0w** [comment]
also feel free to test out my version of this code [here (demo.php)](http://steam-auth.redsparr0w.com/demo.php) and [here (example.php)](http://steam-auth.redsparr0w.com/example.php)

**Comment by bman46** [comment]
Btw the version # is 3.0

**Comment by bman46** [comment]
Make sure u change that on the website view b/c it says 2. Something

**Comment by RedSparr0w** [comment]
Thanks, Updated [=

**Comment by SmItH197** [comment]
1869d2cdaa96a18c3bd11d58770c76bf0d2876c0 created a conflict in the README file, I've fixed this and merged.

**Pull request merged**